reactive/src/main/java/feign/reactive/ReactiveDelegatingContract.java

```diff
@@ -55,7 +57,12 @@ public List<MethodMetadata> parseAndValidateMetadata(Class<?> targetType) {
           throw new IllegalArgumentException(
               "Streams are not supported when using Reactive Wrappers");
         }
-        metadata.returnType(actualTypes[0]);
+        if (isReactiveOf(type, Flux.class)) {
+          ParameterizedType wrappedType = TypeUtils.parameterize(List.class, actual);
```

Comment:
I'm not sure swapping the type from under the user is safe, I think the decoder should know how to deal with it.

Imagine we are talking about `LocalDateTime`.  Feign won't change it to `java.util.Date` so decoders do their trick and later change the type back to `LocalDateTime`.

I don't think this change is safe for feign

Reply:
> I'm not sure swapping the type from under the user is safe, I think the decoder should know how to deal with it.
> 
> Imagine we are talking about `LocalDateTime`. Feign won't change it to `java.util.Date` so decoders do their trick and later change the type back to `LocalDateTime`.
> 
> I don't think this change is safe for feign

But we've already done this here, we change the type `Publisher` type to the normal one. It's a delegation class, it's meant to do this. So why not distinguish the `Publisher` type further. The `Mono` and `Flux` definitely have different semantics, as I mentioned [here](https://github.com/OpenFeign/feign/issues/1869#issuecomment-1336363851).

The best solution is to implement the real reactive decoder, but currently we just need a wrapper.